src/proto/uprotocol/upayload.rs

```diff
@@ -14,10 +14,66 @@
 use prost::Message;
 use prost_types::Any;
 
-use crate::{
-    cloudevent::serializer::SerializationError,
-    uprotocol::{Data, UPayload, UPayloadFormat},
-};
+pub use crate::types::serializationerror::SerializationError;
+use crate::uprotocol::{Data, UPayload, UPayloadFormat};
+
+const MIME_SUBSTYPE_JSON: &str = "json";
+const MIME_SUBTYPE_PROTOBUF: &str = "x-protobuf";
+const MIME_SUBTYPE_RAW: &str = "octet-stream";
+const MIME_SUBTYPE_SOMEIP: &str = "x-someip";
+const MIME_SUBTYPE_SOMEIP_TLV: &str = "x-someip_tlv";
+const MIME_SUBTYPE_PLAIN: &str = "plain";
+
+impl UPayloadFormat {
+    /// Gets the payload format that corresponds to a given MIME type.
+    ///
+    /// # Returns
+    ///
+    /// The corresponding payload format or [`UPayloadFormat::UpayloadFormatProtobuf`] if the MIME
```

Comment:
Could you elaborate a little on why we'd want to default to `UPayloadFormat::UpayloadFormatProtobuf`?

A natural default would seem to me to be `UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED`.

Perhaps I'm missing something.

Reply:
My understanding is that when serializing a UMessage, a default of Protobuf encoding should be assumed, if no specific format has been set explicitly, because in 98% of all cases that will be the format being used.

@stevenhartley maybe you can shed more light on this?

Reply:
Gotcha, so we hang tight till that PR merges and then we can revisit and update the logic here to match what you wrote.